fineract-core/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java

```diff
@@ -443,51 +430,32 @@ private void validateNameParts(final List<ApiParameterError> dataValidationError
         }
     }
 
-    private void validateActivationDate(final List<ApiParameterError> dataValidationErrors) {
+    private void validateActivationDate() {
 
         if (getSubmittedOnDate() != null && isDateInTheFuture(getSubmittedOnDate())) {
-
-            final String defaultUserMessage = "submitted date cannot be in the future.";
-            final ApiParameterError error = ApiParameterError.parameterError("error.msg.clients.submittedOnDate.in.the.future",
-                    defaultUserMessage, ClientApiConstants.submittedOnDateParamName, this.submittedOnDate);
-
-            dataValidationErrors.add(error);
+            throw new InvalidClientActivationDateException("submittedOnDate.in.the.future", "submitted date cannot be in the future.",
+                    getSubmittedOnDate());
         }
 
         if (getActivationLocalDate() != null && getSubmittedOnDate() != null && getSubmittedOnDate().isAfter(getActivationLocalDate())) {
-
-            final String defaultUserMessage = "submitted date cannot be after the activation date";
-            final ApiParameterError error = ApiParameterError.parameterError("error.msg.clients.submittedOnDate.after.activation.date",
-                    defaultUserMessage, ClientApiConstants.submittedOnDateParamName, this.submittedOnDate);
-
-            dataValidationErrors.add(error);
+            throw new InvalidClientActivationDateException("submittedOnDate.after.activation.date",
+                    "submitted date cannot be after the activation date", getSubmittedOnDate());
         }
 
         if (getReopenedDate() != null && getActivationLocalDate() != null && getReopenedDate().isAfter(getActivationLocalDate())) {
-
-            final String defaultUserMessage = "reopened date cannot be after the submittedon date";
-            final ApiParameterError error = ApiParameterError.parameterError("error.msg.clients.submittedOnDate.after.reopened.date",
-                    defaultUserMessage, ClientApiConstants.reopenedDateParamName, this.reopenedDate);
-
-            dataValidationErrors.add(error);
+            throw new InvalidClientActivationDateException("submittedOnDate.after.reopened.date",
```

Comment:
Wrong type of error: The problem is not with the activate date.
this might break the contract. Priorly the problematic field name was sent, i reckon.